ops/jujuversion.py

```diff
@@ -0,0 +1,63 @@
+import re
+from functools import total_ordering
+
+
+@total_ordering
+class JujuVersion:
+
+    PATTERN = r'^(?P<major>\d{1,9})\.(?P<minor>\d{1,9})((?:\.|-(?P<tag>[a-z]+))(?P<patch>\d{1,9}))?(\.(?P<build>\d{1,9}))?$'
```

Comment:
Why 1-9, and not simply +?  There's nothing special about 9 digits in this context, so this is shorter and more correct.

Reply:
I was following the Juju code base in that regard:
https://github.com/juju/version/blob/81c1be00b9a67de4490453f6117d13d818c9fb84/version.go#L108

I could change it but it would be inconsistent with `juju/version`.